Amplitude/Amplitude.m

```diff
@@ -965,16 +971,15 @@ - (void)setUserProperties:(NSDictionary*) userProperties replace:(BOOL) replace
 
 - (void)setUserId:(NSString*) userId
 {
-    if (![self isArgument:userId validType:[NSString class] methodName:@"setUserId:"]) {
+    if (!([self isArgument:userId validType:[NSString class] methodName:@"setUserId:"] || userId == nil)) {
         return;
     }
     
     [self runOnBackgroundQueue:^{
-        (void) SAFE_ARC_RETAIN(userId);
         SAFE_ARC_RELEASE(_userId);
-        _userId = userId;
+        _userId = SAFE_ARC_RETAIN(userId);
```

Comment:
I think the retain has to come before the release, otherwise userId could be freed before this is set. could be wrong though